public/data/cpp.json

```diff
@@ -39,6 +39,86 @@
         ],
         "tags": ["cpp", "string", "split", "utility"],
         "author": "saminjay"
+      },
+      {
+        "title": "Convert to Uppercase",
+        "description": "Converts all characters in a string to uppercase.",
+        "code": [
+          "#include <string>",
+          "#include <algorithm>",
+          "",
+          "std::string toUpperCase(const std::string& input) {",
```

Comment:
Why are you taking the ref by reference ?

Reply:
I think a constant reference is a good thing. It will not copy the whole variable, but just create a reference. But in this case yes, instead of making a const reference here and then creating another copy to return the string. They can just pass by value and return that string after transforming.